app/src/main/java/richard/chard/lu/android/areamapper/activities/BaseActivity.java

```diff
@@ -62,7 +69,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         LOG.trace("Entry");
+
+        MapsInitializer.initialize(getApplicationContext(), Renderer.LATEST, null);
```

Comment:
curious if `AreaMapperActivity` `onCreate` is a better place for this method as `BaseActivity` doesn't even have a map view ?

Reply:
good point. This initialization happens only once per app lifecycle and only the first call takes effect, so to force the use of the new renderer it needs to be called before MapView.onCreate(). The documentation suggests putting in the Application.onCreate() but because we don't have a custom Application class, I thought about calling it from the base activity just to have a logical separation between this call and the actual Map view initialization. Tests also revealed that both options work fine.

Reply:
Got it. I think putting it in `BaseActivity` is not different from putting it in `AreamapperActivity` at the moment, though if we create another activity tomorrow extending `BaseActivity` , this call would result in happening twice for each of the activities extending `BaseActivity`. IMO adding a custom Application class and puting it in `Application.onCreate()` is probably the best design here. That will also give us a better place for handling these once per application life cycle events in future.

Reply:
I completely agree that that's a better design. I was just not convinced that if it was justifiable to adopt it to initialize an sdk but your comment motivated me.